engine/src/main/java/org/pentaho/di/job/entries/ping/JobEntryPing.java

```diff
@@ -198,7 +198,26 @@ public void saveRep( Repository rep, IMetaStore metaStore, ObjectId id_job ) thr
     }
   }
 
+  public static boolean isNumeric( String strNum ) {
```

Comment:
If similar functionality is needed in the dialog also, we should try to make this a method used between both somehow.  
Also, we should add a unit test for the new logic, at minimum (and/or make an integration test ensuring the step behaves as desired when provided with bad data).